smfile_parser.py

```diff
@@ -48,56 +47,51 @@ def parse_sm(sm_file, new_file, output_dir):
     measure         = []
     measure_index   = 0
 
-    flag            = False
+    read_notes      = False
 
     with open(sm_file, encoding='ascii', errors='ignore') as f:
         for line in f:
-            if not flag:
+            line = line.rstrip('\n')
+            if not read_notes:
                 if line.startswith('#TITLE:'):
-                    step_dict['title']  = line.lstrip('#TITLE').lstrip(':').rstrip(';\n')
+                    step_dict['title']  = line.lstrip('#TITLE').lstrip(':').rstrip(';')
                 elif line.startswith('#BPMS:'):
                     if ',' in line:  # raises Exception if multiple BPMS detected
                         raise ValueError('Multiple BPMs detected')
-                    step_dict['BPM']    = float(line.lstrip('#BPMS:0.0').lstrip('0').lstrip('=').rstrip(';\n'))
+                    step_dict['bpm']    = float(split('=', line)[-1].rstrip(';'))
                 elif line.startswith('#OFFSET:'):
-                    step_dict['offset'] = float(line.lstrip('#OFFSET').lstrip(':').rstrip(';\n'))
-                elif line.startswith('#STOPS:') and line.rstrip('\n') != "#STOPS:;":
+                    step_dict['offset'] = float(split(':', line)[-1].rstrip(';'))
+                elif line.startswith('#STOPS:') and line != "#STOPS:;":
                     raise ValueError('Stop detected')
-
                 elif line.startswith('#NOTES:'):
-                    flag     = True
+                    read_notes     = True
+
+            if read_notes:   #start of note processing
+                if line.startswith('#NOTES:'):
+                    measure_index = 0
                     next(f)
                     next(f)
-                    step_dict['types'].append("DIFFICULTY ")
-                    step_dict['notes'].append(next(f).lstrip(' ').rstrip(':\n'))
-
-            else:   #start of note processing
-                if line[0].isdigit():
-                    check = True if any((c in set('123456789')) for c in line) else False
-                    if check:
-                        measure.append(True)
-                        step_dict['types'].append(convert_note(line.rstrip('\n')))
-                    else:
-                        measure.append(False)
-                elif line.startswith(',') or line.startswith(';'):
-                    line_timing = calculate_timing(measure, measure_index, step_dict['BPM'], step_dict['offset'])
+                    step_dict['notes'].append('DIFFICULTY ' + next(f).lstrip(' ').rstrip(':\n'))
```

Comment:
It's possible this can cause EOF exceptions if the file is not properly formatted.

Reply:
Also, why add the difficulty to notes? You can make 'notes' be a list of dictionaries with the keys being the difficulty name and the value being the measures.